src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -143,7 +143,7 @@ public string GenerateStartTag()
         /// </returns>
         public string GeneratePreContent()
         {
-            if (SelfClosing)
+            if (!string.IsNullOrWhiteSpace(TagName) && SelfClosing)
```

Comment:
`TagName` value does _not_ change content and should not be referenced here.  `TagName==null` means "generate the content but not the wrapping start and end tags".